junebug/tests/test_utils.py

```diff
@@ -298,3 +299,99 @@ def test_api_from_status(self):
             'message': 'Bar',
             'details': {'baz': 'quux'}
         })
+
+    def test_public_http_properties_explicit(self):
+        result = channel_public_http_properties({
+            'web_path': '/baz/quux',
+            'web_port': 2121,
+            'public_http': {
+                'web_path': '/foo/bar',
+                'web_port': 2323,
+            }
```

Comment:
Some changes I made that is different to what we last discussed:
  - Using `public_http` instead of `nginx`, seems more descriptive of the purpose of this configuration, but happy to use a different name
  - The public http properties are now a sub-property of the channel config given to junebug via a `POST /channels/`, seems better placed there than outside of the config, but happy to move it outside of the config if it would make more sense.
   - Not sure if we discussed this, but we allow any of `web_path` and `web_port` to be included in the channel config, and the same field names in `public_http` just override those if present.